mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.erb

```diff
@@ -4207,4 +4261,277 @@ resource "google_container_node_pool" "without_confidential_boot_disk" {
 }
 `, cluster, networkName, subnetworkName, np)
 }
+
+func testAccContainerNodePool_resourceManagerTags(projectID, clusterName, networkName, subnetworkName, randomSuffix string) string {
+	return fmt.Sprintf(`
+data "google_project" "project" {
+  project_id = "%[1]s"
+}
+
+resource "google_project_iam_binding" "tags" {
+  project = "%[1]s"
+  role    = "roles/resourcemanager.tagHoldAdmin"
+  members = [
+    "serviceAccount:service-${data.google_project.project.number}@container-engine-robot.iam.gserviceaccount.com",
+  ]
+}
+
+resource "google_tags_tag_key" "key1" {
+  parent = "projects/%[1]s"
+  short_name = "foobarbaz1-%[2]s"
+  description = "For foo/bar1 resources"
+  purpose = "GCE_FIREWALL"
+  purpose_data = {
+    network = "%[1]s/%[4]s"
+  }
+}
+
+resource "google_tags_tag_value" "value1" {
+  parent = "tagKeys/${google_tags_tag_key.key1.name}"
+  short_name = "foo1-%[2]s"
+  description = "For foo1 resources"
+}
+
+resource "google_tags_tag_key" "key2" {
+  parent = "projects/%[1]s"
+  short_name = "foobarbaz2-%[2]s"
+  description = "For foo/bar2 resources"
+  purpose = "GCE_FIREWALL"
+  purpose_data = {
+    network = "%[1]s/%[4]s"
+  }
+}
+
+resource "google_tags_tag_value" "value2" {
+  parent = "tagKeys/${google_tags_tag_key.key2.name}"
+  short_name = "foo2-%[2]s"
+  description = "For foo2 resources"
+}
+
+data "google_container_engine_versions" "uscentral1a" {
+  location = "us-central1-a"
+}
+
+resource "google_container_cluster" "primary" {
+  name               = "%[3]s"
+  location           = "us-central1-a"
+  min_master_version = data.google_container_engine_versions.uscentral1a.release_channel_latest_version["STABLE"]
+
+  # We can't create a cluster with no node pool defined, but we want to only use
+  # separately managed node pools. So we create the smallest possible default
+  # node pool and immediately delete it.
+  remove_default_node_pool = true
+  initial_node_count = 1
+
+  deletion_protection = false
+  network             = "%[4]s"
+  subnetwork          = "%[5]s"
+
+  timeouts {
+    create = "30m"
+    update = "40m"
+  }
+}
+
+# Separately Managed Node Pool
+resource "google_container_node_pool" "primary_nodes" {
+  name       = google_container_cluster.primary.name
+  location   = var.region
+  cluster    = google_container_cluster.primary.name
+
+  version = data.google_container_engine_versions.uscentral1a.release_channel_latest_version["STABLE"]
+  node_count = var.gke_num_nodes
+
+  node_config {
+    machine_type    = "n1-standard-1"  // can't be e2 because of local-ssd
+    disk_size_gb    = 15
+
+    resource_manager_tags = {
+      "tagKeys/${google_tags_tag_key.key1.name}" = "tagValues/${google_tags_tag_value.value1.name}"
+    }
+  }
+}
+`, projectID, randomSuffix, clusterName, networkName, subnetworkName)
+}
+
+func testAccContainerNodePool_resourceManagerTagsUpdate1(projectID, clusterName, networkName, subnetworkName, randomSuffix string) string {
```

Comment:
This whole block is within a <unless version.... tag, so it only shows up in the beta provider.